CT-1026 Anonymize database dumps

stephenrichards opened the pull request.

aliuk2012 commented on config/routes.rb:
```diff
@@ -10,5 +10,6 @@
 
   root to: 'correspondence#index'
 
+  get '/search' => 'correspondence#search'
```
Any particular reason why we need a separate endpoint for searching? couldn't we just reuse correspondence index action and test for search param using if params[:search].present?

EdwardAndress replied:
We could and, in fact, we could also just modify #search on the model so that we don't need any logic in the controller's #index action.  I made the choice to do it in this way for reasons that only relate to style.
- A form that submits to the #index action seems a little weird to me
- Having to look at the model, only to find that without params[:search] you're getting #all the records also seems wrong
- Having logic in the index action also feels wrong to me... it should just show you everything, always.
That's just my opinion though and we should chat tomorrow if you feel it's neater to this another way.

stephenrichards committed:
CT-1026 Anonymise Data dump

Database dumps taken from prod are now anonymised. User data with email addresses ending @digital.justice.gov.uk are not anonymised, all other user, requester and requester data is.

Request subjects and messages are prefixed by letters representing the initial letters of the first 10 words of the original, in order to help confirm we are looking at the correct record.

stephenrichards committed:
Remove unneccesary to_s

misaka reviewed (changes requested):
I think we just need an addition to the README for doing data dumps.

misaka reviewed (changes requested):
Moving this back to requested changes. :-/ I've asked @beccapearce to have a look at adding a section to the README

Commit:
CT-1026 updates the readme

Commit:
CT-1026 adds db:dump:help tot he readme
